contracts/currency/currency.go

```diff
@@ -282,12 +298,13 @@ func GetExchangeRate(currencyAddress *common.Address, header *types.Header, stat
 }
 
 // GetBalanceOf returns an account's balance on a given ERC20 currency
-func GetBalanceOf(accountOwner common.Address, contractAddress common.Address, header *types.Header, state vm.StateDB) (result *big.Int, err error) {
+func GetBalanceOf(vmRunner vm.EVMRunner, accountOwner common.Address, contractAddress common.Address) (result *big.Int, err error) {
 	log.Trace("GetBalanceOf() Called", "accountOwner", accountOwner.Hex(), "contractAddress", contractAddress)
 
-	err = contract_comm.MakeStaticCallWithAddress(contractAddress, balanceOfFuncABI, "balanceOf", []interface{}{accountOwner}, &result, params.MaxGasToReadErc20Balance, header, state)
+	err = getBalanceMethod.Bind(contractAddress).Query(vmRunner, &result, accountOwner)
+
 	if err != nil {
-		log.Error("GetBalanceOf evm invocation error", "err", err)
+		log.Error("GetBalanceOf evm invocation error", "leftoverGas", "err", err)
```

Comment:
This would give an error due to the odd number of arguments after the message.  I don't remember the exact error it gives, but I've seen it many times when adding logging to debug something.